app/models/katello/concerns/smart_proxy_extensions.rb

```diff
@@ -129,6 +129,11 @@ def backend_service_type(repository)
       end
 
       def pulp3_support?(repository)
+        # we might be looking at a ContentViewPuppetEnvironment
```

Comment:
Not sure if this is the best place to do this check but seems reasonable. It could also be done in MetadataGenerate but this is slightly more central

Reply:
We could put a check like this: 
https://github.com/Katello/katello/blob/97c0da150c9ab08951e06d13bc585f09a1df708c/app/lib/actions/pulp/repository/distributor_publish.rb#L9
I believe the root check works but I have seen this more often in our code.

Also, I think the metadata generate would be the right place to put this. We shouldn't be passing cvpe to this method and I think the check should be handled before making the call so this method can be as generic as possible.

Reply:
I agree on moving the code on your point about leaving this to be generic. I've never been a fan of checking object types, though. I prefer checking the kinds of messages they can receive. I'll think about that more

Reply:
Ok, updated. Found a better way thanks to how we are delegating:

https://github.com/Katello/katello/blob/1a4660e9ff71165f7ce0c07df62a855ea166ab55/app/models/katello/repository.rb#L151-L152

A Repository and a ContentViewPuppetEnvironment both respond to `content_type`!

Reply:
Perfect! Looks great..Tested cv publish/promote with puppet modules and mixed content. Also tested file create in pulp3. Works great for all the flows I tested.. :+1